include/PassType.h

```diff
@@ -0,0 +1,51 @@
+#ifndef PASS_H
+#define PASS_H
+
+#include "RunAnalysis.h"
+#include "Utils.h"
+#include "PointerAnalysis.h"
+#include "CFG.h"
+
+#include <set>
+#include <fstream>
+
+typedef std::map<std::string, std::string> AliasMap;
+
+struct MethodsSetHolder {
+    std::set<std::string> methodsSet; 
+    bool setInitialized; 
+};
+
+typedef std::map<std::string, std::map<std::string, MethodsSetHolder>> MappedMethods;
+
+class PassType {
+private: 
+    AliasMap aliasedVars; 
+    std::string testName; 
+    CFG* cfg; 
+    MappedMethods expectedResult; 
+    
+    virtual void analyzeCFG(CFG* cfg, MappedMethods& PreMappedMethods, MappedMethods& PostMappedMethods, std::string priorBranch) = 0; 
+    virtual void transfer(Instruction* I,  SetVector<Instruction *> workSet, std::map<std::string, MethodsSetHolder>& inputMethodsSet) = 0;
+public:    
+    void setFunctions(std::map<std::string, bool> safeFunctions, std::map<std::string, bool> unsafeFunctions, std::map<std::string, bool> reallocFunctions, std::map<std::string, std::string> memoryFunctions); 
+    void buildExpectedResult(std::string testName); 
+
+    std::map<std::string, bool> safeFunctions;
+    std::map<std::string, bool> unsafeFunctions; 
+    std::map<std::string, bool> reallocFunctions;
+    std::map<std::string, std::string> memoryFunctions;
+    std::string passName; 
+
+   
+    bool runTests(MappedMethods receivedResult); 
```

Comment:
Why is running the tests something that a pass can do? I would expect `runOnCFG` (or something similar) as a method within the pass, and then have `runTests` be in some other class that serves exclusively as a test runner.

Put another way, it would be better to put the _testing logic_ in its own class, so that it doesn't get mixed in with the logic for just running the analysis.